src/server/services/airtable.ts

```diff
@@ -0,0 +1,506 @@
+import bcrypt from 'bcryptjs';
+
+import db from '@/server/db';
+import base from '@/server/db/airtable';
+import { parentLogger } from '@/server/helpers/logger';
+import { type ApiAccount } from '@/types/ApiAccount';
+import { Airtable } from '@/types/enum/Airtable';
+import { USER_ROLE } from '@/types/enum/UserRole';
+import { diff } from '@/utils/array';
+import { sanitizeEmail } from '@/utils/validation';
+
+import { sendInscriptionEmail } from '../email';
+
+const DRY_RUN = process.env.DRY_RUN === 'true';
+
+const logger = parentLogger.child({
+  dry_run: DRY_RUN,
+});
+
+const logDry: typeof console.log = (message) => logger.info(`${DRY_RUN ? '[DRY_RUN]' : '[LIVE]'} ${message}`);
+
+export interface ApiNetwork {
+  id_sncu: string;
+  full_url: string;
+  public_name: string;
+  contacts: string[];
+}
+
+export const syncGestionnairesWithUsers = async () => {
+  await sanitizeGestionnairesEmails();
+  logDry(`Sync users from "${Airtable.GESTIONNAIRES}" sheet`);
+  const gestionnaires = await base(Airtable.GESTIONNAIRES).select().all();
+  const users = await db('users')
+    .select('id', 'email', 'active', 'gestionnaires', 'receive_new_demands', 'receive_old_demands')
+    // QUESTION should we update only gestionnaires ?
```

Comment:
should we ? C'est toujours d'actualité ?

Reply:
Oui c'était une question pour toi,
je ne suis pas sur des roles dans la table user, donc je ne sais pas si on doit updater uniquement les roles gestionnaires ou si d'autres types de user peuvent aussi être désactivés